### PR TITLE
docs: schema generation

### PR DESCRIPTION
**Pull request by hoffa**
### Description of changes

Some barebones overview of how the schema is generated.

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/schema.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# AWS SAM JSON schema
+
+## Context
+
+The [AWS SAM specification](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-specification.html) is defined by a [JSON schema](https://github.com/aws/serverless-application-model/discussions/2645).
```
Comment (acristin) [inline docs/schema.md:5]: Very non-blocking nit: I think the schama is derived from the SAM code, right? So technically, it isn't defined by the schema.

File: docs/schema.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## Schema generation
+
+At a high level, the final [`schema.json`](https://github.com/aws/serverless-application-model/blob/develop/samtranslator/schema/schema.json) is generated as such:
+
+```mermaid
```
Comment (ssenchenko) [inline docs/schema.md:11]: It'd be great to see a screenshot with an actual diagram

**Review by ssenchenko — APPROVED** [review]

File: docs/schema.md (hunk 2 above)
Comment (hoffa) [reply docs/schema.md:11]: Click on the little document icon in the top bar

**Review by acristin — APPROVED** [review]

File: docs/schema.md (hunk 1 above)
Comment (hoffa) [reply docs/schema.md:5]: Any suggestions on wording?

**Review by aahung — APPROVED** [review]

**Review by xazhao — APPROVED** [review]

File: docs/schema.md (hunk 1 above)
Comment (aahung) [reply docs/schema.md:5]: The @acristin 's "derived" is good. A json schema is derived from SAM spec and xxx

**Pull request merged**